src/cata_tiles.cpp

```diff
@@ -1599,69 +1611,64 @@ void cata_tiles::draw( const point &dest, const tripoint &center, int width, int
         }
     };
 
-    // Limit draw depth to vertical vision setting
-    // Disable multi z-level display on isometric tilesets until height_3d issues resolved
-    const int max_draw_depth = is_isometric() ? 0 : fov_3d_z_range;
     const int height_3d_mult = is_isometric() ? 10 : 0;
     if( max_draw_depth <= 0 ) {
         // Legacy draw mode
-        for( tile_render_info &p : draw_points ) {
+        for( int row = min_row; row < max_row; row ++ ) {
             for( auto f : drawing_layers_legacy ) {
-                ( this->*f )( p.pos, p.ll, p.height_3d, p.invisible );
+                for( tile_render_info &p : draw_points[row] ) {
+                    ( this->*f )( p.pos, p.ll, p.height_3d, p.invisible );
+                }
             }
         }
     } else {
         // Multi z-level draw mode
-
-        // Categorize draw_points by lowest level to draw
-        std::map<int, std::vector<tile_render_info>> draw_points_3d;
-        for( tile_render_info &p : draw_points ) {
-            tripoint p_draw = p.pos;
-            while( !here.dont_draw_lower_floor( p_draw ) && p.pos.z - p_draw.z < max_draw_depth ) {
-                p_draw.z -= 1;
-            }
-            draw_points_3d[p_draw.z].emplace_back( p );
-        }
-
         // Start drawing from the bottom-most z-level
         int cur_zlevel = -OVERMAP_DEPTH;
         do {
             int cur_height_3d = ( cur_zlevel - center.z ) * height_3d_mult;
```

Comment:
I think this should be using the `height_3d` member of `tile_render_info`, because otherwise the 3d height will accumulate across different tiles.

Reply:
It is part of the WIP height_3d changes for isometric 3D that got pushed out early.  Isometric tilesets shouldn't get to that point so it should be harmless, but I could comment it out nevertheless.